crates/ty_python_semantic/src/types.rs

```diff
@@ -9480,6 +9480,8 @@ impl<'db> FieldInstance<'db> {
 pub enum TypeVarKind {
     /// `T = TypeVar("T")`
     Legacy,
+    /// `Alias: typing.TypeAlias = T`
+    Pep613,
```

Comment:
This variant is unlike the others -- it is not really a kind of typevar, but rather a kind of type alias -- a direct alias to a typevar.

I think this is the most direct implementation, but I would probably name and document this variant so as to emphasize its differentness more.